xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -1294,6 +1294,7 @@ void CVideoPlayer::Process()
       CSingleLock lock(m_StateSection);
       if (m_displayLost)
       {
+        CSingleExit unlock(m_StateSection);
```

Comment:
Not sure why it makes no sense. Seems a valid fix that is done similarly in many places in the code.
But I can switch to using atomics for m_displayLost if that is desired.